pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -193,6 +194,10 @@ func (l *LibvirtDomainManager) preStartHook(vm *v1.VirtualMachine, domain *api.D
 			return domain, err
 		}
 	}
+	// create empty disks if they exist
+	if err := emptydisk.CreateTemporaryDisks(vm); err != nil {
```

Comment:
What would you think about having these disks generated before the VM is scheduled on a node as part of the VM Pod's initialization.

We could schedule side car container in the VM pod to provision the empty disks. Virt-controller wouldn't schedule the VM until the disks are available because virt-controller waits for all containers in the POD to become "Ready" first.

My thought process behind this is that I'm uncertain how long it takes to allocate disks if they are huge. We might want to do that before virt-handler beings the startup phase.